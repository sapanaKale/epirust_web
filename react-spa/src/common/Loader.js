import React from 'react';

export default function Loader() {
    return (
        <div id="loader" data-testid="loader" className="multi-spinners">
            <div className="spinner-grow text-dark" />
            <div className="spinner-grow text-dark" />
            <div className="spinner-grow text-dark" />
            <div className="spinner-grow text-dark" />
            <div className="spinner-grow text-dark" />
            <div className="spinner-grow text-dark" />
        </div>
    );
}