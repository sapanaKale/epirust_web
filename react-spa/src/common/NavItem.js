/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import {Link, useRouteMatch} from "react-router-dom";
import React from "react";

export function NavItem({ name, linksTo, activeOnExactMatch = false }) {
  const match = useRouteMatch({
    path: linksTo,
    exact: activeOnExactMatch
  });

  return (
    <li className={`nav-item`}>
      <Link className={`nav-link ${match ? "active" : ""}`} to={linksTo}>{name}</Link>
    </li>
  );
}