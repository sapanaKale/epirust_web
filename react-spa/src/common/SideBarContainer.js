/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import React, {useState} from "react";
import {faTimes} from "@fortawesome/free-solid-svg-icons";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";

export default function SideBarContainer({children}) {
  const [sidebarOpen, updateSideBarOpen] = useState(false);

  return (
    <div className="drawer-wrap">
      <button onClick={() => updateSideBarOpen(status => !status)} type="button" className="btn btn-primary" aria-label="Close">
        <span>Settings</span>
        <svg className="bi bi-gear-fill" width="1em" height="1em" viewBox="0 0 16 16" fill="currentColor"
             xmlns="http://www.w3.org/2000/svg">
          <path fillRule="evenodd"
                d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 01-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 01.872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 012.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 012.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 01.872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 01-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 01-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 100-5.86 2.929 2.929 0 000 5.858z"
                clipRule="evenodd"/>
        </svg>
      </button>
      <div className={`drawer ${sidebarOpen && "open"}`}>
        <FontAwesomeIcon onClick={() => updateSideBarOpen(status => !status)} icon={faTimes}/>
        {children}
      </div>
    </div>
  );
}
