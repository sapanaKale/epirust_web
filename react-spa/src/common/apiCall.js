import config from "../config";

export const post = (url, data) => {
  return fetch(`${config.API_HOST}/api${url}`, {
    method: 'POST',
    headers: {'Content-Type': 'application/json'},
    body: JSON.stringify(data)
  })
    .then(response => handleNon200(response))
}

export const get = (url) => {
  return fetch(`${config.API_HOST}/api${url}`, {
    method: 'GET',
    headers: {'Content-Type': 'application/json'}
  })
    .then(response => handleNon200(response))
}

const handleNon200 = (response) => {
  if (!response.ok) {
    throw Error(response.statusText);
  }
  return response;
}