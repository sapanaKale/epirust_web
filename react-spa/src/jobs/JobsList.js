/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import {JobTile} from "./JobTile";
import React, {useEffect, useState} from "react";
import TagsSelect from "../simulation-configuration/TagsSelect";
import {useHistory, useLocation} from "react-router-dom";

const REACT_SELECT_ACTIONS = {
  CLEAR: 'clear',
  SELECT: 'select-option',
  POP: 'pop-value',
  REMOVE: 'remove-value'
};


export const JobsList = ({jobs, activeJob}) => {
  let history = useHistory();
  const location = useLocation();

  const [tagIdsToFilter, updateTagIdsToFilter] = useState([]);
  const [listHeight, updateListHeight] = useState(0);
  const filterFormRef = React.useRef(null);

  useEffect(() => {
    if (!location)
      return;

    let filterTags = location.search
      .match(/filterTags=.+?(?=&|$)/);

    if (!filterTags)
      return;

    const filterTagsUrl = filterTags && filterTags[0]
      .split("=")[1]
      .split(',');

    updateTagIdsToFilter(filterTagsUrl)

  }, [location]);

  function handleSelectChange(changeValues, {action}) {
    const changedIds = changeValues ? changeValues.map(option => option.value) : [];

    const filterQueryParams = changedIds.length > 0
      ? `?filterTags=${changedIds.join(',')}`
      : "";

    switch (action) {
      case REACT_SELECT_ACTIONS.SELECT:
      case REACT_SELECT_ACTIONS.CLEAR:
      case REACT_SELECT_ACTIONS.POP:
      case REACT_SELECT_ACTIONS.REMOVE:
        updateTagIdsToFilter(changedIds);
        history.push(`${location.pathname}${filterQueryParams}`);
        break;
    }
  }

  useEffect(() => {
    if (!filterFormRef)
      return;

    const parentElementHeight = filterFormRef.current.parentElement.clientHeight,
      formHeight = filterFormRef.current.clientHeight;

    updateListHeight(parentElementHeight - formHeight)
  }, [filterFormRef, tagIdsToFilter]);

  function isJobTagged(job) {
    if (!tagIdsToFilter || !tagIdsToFilter.length)
      return true;
    return job.config.tags.some(({id}) => tagIdsToFilter.includes(id));
  }

  function handleJobClick(jobId) {
    const filterQueryParams = tagIdsToFilter.length > 0
      ? `?filterTags=${tagIdsToFilter.join(',')}`
      : "";

    history.push(`/jobs/${jobId}${filterQueryParams}`)
  }

  return (
    <>
      <form ref={filterFormRef} className="job-filter-controls" onSubmit={e => e.preventDefault()}>
        <TagsSelect
          label={"Filter"}
          placeholder={"Type tag name..."}
          onChange={handleSelectChange}
          tagIdsSelected={tagIdsToFilter}/>
      </form>
      <ul className="list-group scrollable job-list" data-testid="job-list" style={{maxHeight: listHeight}}>
        {jobs
          .filter(isJobTagged)
          .map(job =>
            <JobTile
              key={job._id}
              jobId={job._id}
              status={job.status}
              tags={job.config.tags}
              isActive={activeJob ? job._id === activeJob._id : false}
              onClick={handleJobClick}
            />)
        }
      </ul>
    </>
  );
};
