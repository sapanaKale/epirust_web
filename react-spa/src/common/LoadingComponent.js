/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import {LOADING_STATES} from "./constants";
import Loader from "./Loader";
import React from "react";
import PropTypes from 'prop-types'

export const LoadingComponent = (props) => {
  switch (props.loadingState) {
    case LOADING_STATES.FINISHED:
      return props.children;

    case LOADING_STATES.FAILED:
      return <div>Failed to load</div>

    default:
      return <Loader/>
  }
}

LoadingComponent.propTypes = {
  loadingState: PropTypes.oneOf([1, 2, 3]).isRequired
}