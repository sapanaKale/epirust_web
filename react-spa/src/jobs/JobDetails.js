/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import React from "react";
import "../styles/components/jobs/_job-details.scss"
import {NavItem} from "../common/NavItem";
import {Redirect, Route, Switch} from "react-router-dom";
import GridPage from "../grid";
import PropTypes from 'prop-types';
import {TimeSeries} from "../time-series";
import Tags from "./Tags";

export const JobDetails = ({jobId, config}) => {
  const linkPrefix = `/jobs/${jobId}`;
  const gridEnabled = config.enable_citizen_state_messages;

  function renderGraphTabs() {
    return (
      <div className="row">
        <div className="col-8">
          <ul className="nav nav-tabs nav-fill">

            <NavItem name="Time Series" linksTo={`${linkPrefix}/time-series`} activeOnExactMatch={true}/>
            {gridEnabled && <NavItem name="Grid" linksTo={`${linkPrefix}/grid`}/>}
            <NavItem name="Config" linksTo={`${linkPrefix}/config`}/>

          </ul>
        </div>

        <div className="col-4">
          <div className="job-meta">
            <p>{`Job: `}<code>{jobId}</code></p>
            <Tags tags={config.tags}/>
          </div>
        </div>
      </div>
    );
  }

  function renderContentForTab() {
    return (
      <div className='job-details-content'>
        <Switch>

          <Route exact path={"/jobs/:id/time-series"}>
            <TimeSeries jobId={jobId}/>
          </Route>

          <Route exact path={"/jobs/:id/grid"}>
            {gridEnabled ? <GridPage jobId={jobId}/> : <Redirect to={`/jobs/${jobId}/time-series`}/>}
          </Route>

          <Route exact path={"/jobs/:id/config"}>
            {<pre>{JSON.stringify(config, undefined, 4)}</pre>}
          </Route>
        </Switch>
      </div>
    );
  }


  return (
    <div className="job-details">
      {renderGraphTabs()}
      {renderContentForTab()}
    </div>
  )
};

JobDetails.propTypes = {
  jobId: PropTypes.string.isRequired,
  config: PropTypes.object.isRequired
};
