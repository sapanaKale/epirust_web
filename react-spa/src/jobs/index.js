/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import React, {useEffect, useState} from "react";
import '../styles/components/jobs/_jobs-list.scss'
import {LOADING_STATES} from "../common/constants";
import {JobsContainer} from "./JobsContainer";
import {LoadingComponent} from "../common/LoadingComponent";
import {JobsRefresher} from "./JobsRefresher";

export const JobsView = () => {
  const [loadingState, updateLoadingState] = useState(LOADING_STATES.LOADING);
  const [jobs, updateJobs] = useState([]);

  useEffect(() => {
    const jobsRefresher = new JobsRefresher(updateJobs, updateLoadingState);
    jobsRefresher.start()
  }, []);

  return <LoadingComponent loadingState={loadingState}>
    <JobsContainer jobs={jobs}/>
  </LoadingComponent>;
};
