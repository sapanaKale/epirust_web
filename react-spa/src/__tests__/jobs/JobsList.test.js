/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import {JobsList} from "../../jobs/JobsList";
import React from "react";
import {render} from "@testing-library/react";
import {MemoryRouter} from "react-router-dom";

import selectEvent from "react-select-event";
import {get} from "../../common/apiCall";
import {renderWithRouter} from "../helper/router-helper";

jest.mock("../../common/apiCall");

describe('JobsList', () => {
  let sarsTag = {id: "12", name: "SARS"}, covidTag = {id: "11", name: "covid-19"};
  let sarsJobId = "12345", covidJobId = "12344", sarsCovidJob = "12346", noTagsJob = '123';

  const testJobs = [
    {_id: covidJobId, status: "finished", config: {tags: [covidTag]}},
    {_id: sarsJobId, status: "finished", config: {tags: [sarsTag]}},
    {_id: sarsCovidJob, status: "finished", config: {tags: [covidTag, sarsTag]}},
    {_id: noTagsJob, status: "finished", config: {tags: []}}
  ];

  get.mockImplementation((url) => {
    if (/jobs\/tags/.test(url))
      return Promise.resolve({json: jest.fn().mockResolvedValueOnce([])});

    return Promise.resolve({json: jest.fn().mockResolvedValueOnce({})});
  });

  it('should render jobs with active job if valid active job', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <JobsList jobs={testJobs} activeJob={testJobs[1]}/>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot()
  });

  it('should render jobs list with no active active job if it is invalid', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <JobsList jobs={testJobs} activeJob={null}/>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot()
  });

  //TODO: should render jobs list with no active active job if it is invalid

  it('should filter elements by selected tags', async () => {

    get.mockImplementation((url) => {
      if (/jobs\/tags/.test(url))
        return Promise.resolve({json: jest.fn().mockResolvedValueOnce([covidTag, sarsTag])});

      return Promise.resolve({json: jest.fn().mockResolvedValueOnce({})});
    });

    const {getByTestId, getByLabelText, getByText} = render(
      <MemoryRouter>
        <JobsList jobs={testJobs} activeJob={testJobs[1]}/>
      </MemoryRouter>
    );

    //display all by default
    expect(getByTestId('job-list').children.length).toBe(4)

    //display for selected tag
    await selectEvent.select(getByLabelText('Filter'), [sarsTag.name]);
    expect(getByTestId('job-list').children.length).toBe(2)
    expect(getByText(sarsJobId)).toBeInTheDocument();
    expect(getByText(sarsCovidJob)).toBeInTheDocument();

    await selectEvent.select(getByLabelText('Filter'), [covidTag.name]);
    expect(getByTestId('job-list').children.length).toBe(3);
    expect(getByText(sarsJobId)).toBeInTheDocument();
    expect(getByText(sarsCovidJob)).toBeInTheDocument();
    expect(getByText(covidJobId)).toBeInTheDocument();

    //display all when cleared tags
    await selectEvent.clearAll(getByLabelText('Filter'));
    expect(getByTestId('job-list').children.length).toBe(4)
  });

  it('should filter elements by the tags present in the URL on load and later on user selection', async () => {

    get.mockImplementation((url) => {
      if (/jobs\/tags/.test(url))
        return Promise.resolve({json: jest.fn().mockResolvedValueOnce([covidTag, sarsTag])});

      return Promise.resolve({json: jest.fn().mockResolvedValueOnce({})});
    });

    const {getByTestId, getByLabelText, getByText} = renderWithRouter(
      <JobsList jobs={testJobs} activeJob={testJobs[1]}/>, {initialEntries: [`/jobs/123/time-series?filterTags=${sarsTag.id}`]}
    );

    expect(getByTestId('job-list').children.length).toBe(2)
    expect(getByText(sarsJobId)).toBeInTheDocument();
    expect(getByText(sarsCovidJob)).toBeInTheDocument();

    await selectEvent.select(getByLabelText('Filter'), [covidTag.name]);
    expect(getByTestId('job-list').children.length).toBe(3);
    expect(getByText(sarsJobId)).toBeInTheDocument();
    expect(getByText(sarsCovidJob)).toBeInTheDocument();
    expect(getByText(covidJobId)).toBeInTheDocument();
  });
});
