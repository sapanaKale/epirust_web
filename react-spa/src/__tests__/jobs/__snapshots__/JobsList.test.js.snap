// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsList should render jobs list with no active active job if it is invalid 1`] = `
<DocumentFragment>
  <form
    class="job-filter-controls"
  >
    <div
      class="input-control tags-wrap"
    >
      <label
        class="font-weight-bold col-form-label-sm"
        for="tag-input"
      >
        Filter
      </label>
      <div
        class=" css-2b097c-container"
      >
        <div
          class=" css-uddrns-control"
        >
          <div
            class=" css-g1d714-ValueContainer"
          >
            <div
              class=" css-q70w24-placeholder"
            >
              Type tag name...
            </div>
            <div
              class="css-drs5yf-Input"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  aria-label="tags"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="tag-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <div
              aria-hidden="true"
              class=" css-at12u2-loadingIndicator"
            >
              <span
                class="css-1yvy2vo-LoadingDot"
              />
              <span
                class="css-zoievk-LoadingDot"
              />
              <span
                class="css-x748d8-LoadingDot"
              />
            </div>
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-6q0nyr-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div>
          <input
            name="tags"
            type="hidden"
          />
        </div>
      </div>
    </div>
  </form>
  <ul
    class="list-group scrollable job-list"
    data-testid="job-list"
    style="max-height: 0;"
  >
    <li
      class="list-group-item borderless item-less-padding"
    >
      <div
        class="simulation-tab  finished"
      >
        <div
          class="card-body card-body-less-padding"
        >
          <p
            class="card-title title"
          >
            <code>
              12344
            </code>
          </p>
          <p
            class="job-status"
            data-testid="job-status-12344"
          >
            Finished
          </p>
          <ul
            class="tags small"
            data-testid="tags"
          >
            <li
              data-testid="tag-11"
            >
              covid-19
            </li>
          </ul>
        </div>
      </div>
    </li>
    <li
      class="list-group-item borderless item-less-padding"
    >
      <div
        class="simulation-tab  finished"
      >
        <div
          class="card-body card-body-less-padding"
        >
          <p
            class="card-title title"
          >
            <code>
              12345
            </code>
          </p>
          <p
            class="job-status"
            data-testid="job-status-12345"
          >
            Finished
          </p>
          <ul
            class="tags small"
            data-testid="tags"
          >
            <li
              data-testid="tag-12"
            >
              SARS
            </li>
          </ul>
        </div>
      </div>
    </li>
    <li
      class="list-group-item borderless item-less-padding"
    >
      <div
        class="simulation-tab  finished"
      >
        <div
          class="card-body card-body-less-padding"
        >
          <p
            class="card-title title"
          >
            <code>
              12346
            </code>
          </p>
          <p
            class="job-status"
            data-testid="job-status-12346"
          >
            Finished
          </p>
          <ul
            class="tags small"
            data-testid="tags"
          >
            <li
              data-testid="tag-11"
            >
              covid-19
            </li>
            <li
              data-testid="tag-12"
            >
              SARS
            </li>
          </ul>
        </div>
      </div>
    </li>
    <li
      class="list-group-item borderless item-less-padding"
    >
      <div
        class="simulation-tab  finished"
      >
        <div
          class="card-body card-body-less-padding"
        >
          <p
            class="card-title title"
          >
            <code>
              123
            </code>
          </p>
          <p
            class="job-status"
            data-testid="job-status-123"
          >
            Finished
          </p>
          <ul
            class="tags small"
            data-testid="tags"
          />
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`JobsList should render jobs with active job if valid active job 1`] = `
<DocumentFragment>
  <form
    class="job-filter-controls"
  >
    <div
      class="input-control tags-wrap"
    >
      <label
        class="font-weight-bold col-form-label-sm"
        for="tag-input"
      >
        Filter
      </label>
      <div
        class=" css-2b097c-container"
      >
        <div
          class=" css-uddrns-control"
        >
          <div
            class=" css-g1d714-ValueContainer"
          >
            <div
              class=" css-q70w24-placeholder"
            >
              Type tag name...
            </div>
            <div
              class="css-drs5yf-Input"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  aria-label="tags"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="tag-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <div
              aria-hidden="true"
              class=" css-at12u2-loadingIndicator"
            >
              <span
                class="css-1yvy2vo-LoadingDot"
              />
              <span
                class="css-zoievk-LoadingDot"
              />
              <span
                class="css-x748d8-LoadingDot"
              />
            </div>
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-6q0nyr-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div>
          <input
            name="tags"
            type="hidden"
          />
        </div>
      </div>
    </div>
  </form>
  <ul
    class="list-group scrollable job-list"
    data-testid="job-list"
    style="max-height: 0;"
  >
    <li
      class="list-group-item borderless item-less-padding"
    >
      <div
        class="simulation-tab  finished"
      >
        <div
          class="card-body card-body-less-padding"
        >
          <p
            class="card-title title"
          >
            <code>
              12344
            </code>
          </p>
          <p
            class="job-status"
            data-testid="job-status-12344"
          >
            Finished
          </p>
          <ul
            class="tags small"
            data-testid="tags"
          >
            <li
              data-testid="tag-11"
            >
              covid-19
            </li>
          </ul>
        </div>
      </div>
    </li>
    <li
      class="list-group-item borderless item-less-padding"
    >
      <div
        class="simulation-tab active shadow finished"
      >
        <div
          class="card-body card-body-less-padding"
        >
          <p
            class="card-title title"
          >
            <code>
              12345
            </code>
          </p>
          <p
            class="job-status"
            data-testid="job-status-12345"
          >
            Finished
          </p>
          <ul
            class="tags small"
            data-testid="tags"
          >
            <li
              data-testid="tag-12"
            >
              SARS
            </li>
          </ul>
        </div>
      </div>
    </li>
    <li
      class="list-group-item borderless item-less-padding"
    >
      <div
        class="simulation-tab  finished"
      >
        <div
          class="card-body card-body-less-padding"
        >
          <p
            class="card-title title"
          >
            <code>
              12346
            </code>
          </p>
          <p
            class="job-status"
            data-testid="job-status-12346"
          >
            Finished
          </p>
          <ul
            class="tags small"
            data-testid="tags"
          >
            <li
              data-testid="tag-11"
            >
              covid-19
            </li>
            <li
              data-testid="tag-12"
            >
              SARS
            </li>
          </ul>
        </div>
      </div>
    </li>
    <li
      class="list-group-item borderless item-less-padding"
    >
      <div
        class="simulation-tab  finished"
      >
        <div
          class="card-body card-body-less-padding"
        >
          <p
            class="card-title title"
          >
            <code>
              123
            </code>
          </p>
          <p
            class="job-status"
            data-testid="job-status-123"
          >
            Finished
          </p>
          <ul
            class="tags small"
            data-testid="tags"
          />
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
